src/lib.rs

```diff
@@ -488,22 +497,46 @@ mod test {
     assert_eq!(data, xor(&xor(&data,&pad), &pad));
   }
 
+
+
   #[test]
-  fn check_write() {
+  fn check_write_and_read() {
     let mut my_storage = MyStorage::new();
+    let size = 3 as u64;
+    let the_string = random_string(size);
+    let mut data_map = datamap::DataMap::None;
+  {  
     let mut se = SelfEncryptor::new(&mut my_storage as &mut Storage, datamap::DataMap::None);
-    let the_string = random_string(3);
     se.write(&the_string, 5u64);
-    assert_eq!(se.file_size, 8u64);
-    let data_map = se.close();
+    assert_eq!(se.file_size, (size + 5) as u64);
```

Comment:
Hi ben,
We have a position offset.
Best wishes
On 25 Mar 2015 18:39, "Benjamin Bollen" <notifications@github.com> wrote:

> In src/lib.rs
> <https://github.com/dirvine/self_encryption/pull/17#discussion_r27152235>:
>
> >      se.write(&the_string, 5u64);
> > -    assert_eq!(se.file_size, 8u64);
> > -    let data_map = se.close();
> > +    assert_eq!(se.file_size, (size + 5) as u64);
>
> Why is this +5?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dirvine/self_encryption/pull/17/files#r27152235>.
>